presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PushdownUtils.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.pinot;
+
+import com.facebook.presto.spi.block.SortOrder;
+import com.facebook.presto.spi.plan.AggregationNode;
+import com.facebook.presto.spi.plan.TopNNode;
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Optional;
+
+public class PushdownUtils
```

Comment:
Let's call this `PinotPushdownUtils`

Reply:
Hmm ... something to discuss about: I want to use these for presto-aresdb connector too. We originally had them in presto-spi as a part of our fork. Is there a designated common module that connectors can share code in ? Or I can always have the presto-aresdb-toolkit depend on presto-pinot-toolkit but that would look weird.

It would be funny to create another module named presto-pinot-aresdb-utils to house these :)

But I can rename them to something pinot specific for now but this is generic.